particle-iot/device-os
# [boron|bsom] u-blox R510 support

# Feature

- u-blox R510 base modem support for Gen 3 Boron & B SoM

# Details

- This branch is based on `feature/SARA-R510-NCP-FW-update` but removes FOTA features

# Testing

- Modem can be on 3.14 ES or 3.15 IP firmware with EVT3 hardware (SARA-R510S-01B-00).
- For B SoM EVT3 hardware and testing, it is necessary to apply this diff and use the SIM INTERNAL setting even if you are using an external SIM with the jumper soldered.
- Run `tinker-serial1-debugging` and switch between Kore AT&T and EtherSIM
- Try cold boot and warm boot, modem should register quickly with 3.15, but may have some delays with 3.14.
- Run `TEST=wiring/no_fixture_cellular` and `TEST=wiring/no_fixture_long_running`
- Run Dioptra tests

```
diff --git a/hal/network/ncp_client/sara/sara_ncp_client.cpp b/hal/network/ncp_client/sara/sara_ncp_client.cpp
index 5f20336c5..f8e6d90c9 100644
--- a/hal/network/ncp_client/sara/sara_ncp_client.cpp
+++ b/hal/network/ncp_client/sara/sara_ncp_client.cpp
@@ -1254,7 +1254,11 @@ int SaraNcpClient::selectSimCard(ModemState& state) {
                 // NOTE: [ch76449] R510S will not retain GPIO's HIGH after a cold boot
                 // Workaround: Set pin that needs to be HIGH to mode "Module status indication",
                 //             which will be set HIGH when the module is ON, and LOW when it's OFF.
-                const int internalSimMode = 10; // Module status indication mode
+                // DO NOT COMMIT
+                // DO NOT COMMIT
+                const int internalSimMode = 255;
+                // DO NOT COMMIT
+                // DO NOT COMMIT
                 if (mode != internalSimMode) {
                     const int r = CHECK_PARSER(parser_.execCommand("AT+UGPIOC=%u,%d",
                             UBLOX_NCP_SIM_SELECT_PIN, internalSimMode));
 ```
